Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/FarePolicy.hs

```diff
@@ -87,8 +87,8 @@ getFarePolicyOnEndRide mbFromLocation toLocationLatLong merchantOpCityId tripCat
   dropSpecialLocation <- Esq.runInReplica (QSpecialLocation.findSpecialLocationByLatLong' toLocationLatLong) >>= mapM (FareProduct.getDropSpecialLocation merchantOpCityId)
   let isDropSpecialLocation = (Just . SL.Drop . DSpecialLocation.id . fst) =<< dropSpecialLocation
   selectedFarePolicy <-
-    if area /= isDropSpecialLocation
-      then getFarePolicy mbFromLocation merchantOpCityId (fromMaybe False isDashboardRequest) tripCategory vehicleServiceTier area mbBookingStartTime txnId
+    if area /= isDropSpecialLocation && isJust isDropSpecialLocation
```

Comment:
area : Just Default, Just Pickup <specialLocationLocation>, Just Drop <specialLocationLocation>
isDropSpecialLocation : Nothing, Just Drop <specialLocationLocation>
**Possible Case**
Just Default -> Just Drop <specialLocationLocation>
Just Drop <specialLocationLocation> -> Just Drop <specialLocationLocation> (id specialLoationId is different)
_ -> nope